### ISSUE TITLE
[2009-07-17] version numbering in different modules (/sapphire/filesystem/upload.php etc.

### ISSUE THREAD

**Issue by silverstripe-issues** [silverstripe/silverstripe-framework#1906]
created by: wwillm
created at: 2009-07-17
original ticket: http://open.silverstripe.org/ticket/4400

---

In different parts I noticed the same versioning idea for uploaded files.
This code is used in some places:

```
$relativeFilePath = ereg_replace('[0-9]*(\.[^.]+$)',$i . '\\1', $relativeFilePath);
```

this works where relativeFilePath contains well behaved patterns, as myfile.pdf, myfile-55.pdf etc.
But if the filename contains only numbers, e.g. "01234.pdf", the new filename is "2.pdf"

This is quite annoying for users!

**Comment by kinglozzer** [comment]
Fixed in master by #4251

**Comment by tractorcow** [comment]
This was also refactored into a self-contained generator class: See https://github.com/silverstripe/silverstripe-framework/blob/1bcd755120f6a6bc74d6dc578a9fe42674394bd3/filesystem/storage/DefaultAssetNameGenerator.php :D
